standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -5275,38 +5275,67 @@ private void removeUnusedColumnDescriptor(MColumnDescriptor oldCD) {
       return;
     }
 
-    boolean success = false;
     Query query = null;
+    Query query2 = null;
+    boolean success = false;
+    LOG.debug("execute removeUnusedColumnDescriptor");
+    DatabaseProduct dbProduct = DatabaseProduct.determineDatabaseProduct(MetaStoreDirectSql.getProductName(pm), conf);
 
+    /**
+     * In order to workaround oracle not supporting limit statement caused performance issue, HIVE-9447 makes
```

Comment:
Maybe this comment could go to the new method where we decide if we can remove the CD or not

Reply:
OK, will rework the code.

Reply:
Thanks